app/controllers/authentication_controller.rb

```diff
@@ -32,11 +32,12 @@ def callback
 
     render json: {
       govuk_account_session: to_account_session(
-        access_token: oauth_response[:access_token],
-        refresh_token: oauth_response[:refresh_token],
+        access_token: oauth_response.fetch(:access_token),
+        refresh_token: oauth_response.fetch(:refresh_token),
+        level_of_authentication: oauth_response.fetch(:result).fetch("level_of_authentication"),
       ),
       redirect_path: redirect_path,
-      ga_client_id: oauth_response[:result]["_ga"],
+      ga_client_id: oauth_response.fetch(:result)["_ga"],
```

Comment:
Is this because with want a KeyError if `:result` isn't in the oauth_response hash?

Reply:
Yeah, a missing `oauth_response[:access_token]`, `:response_token`, or `:result` would be a bug, so I thought to use `.fetch` to make that really clear.

Reply:
Is that not as important in `app/controllers/attributes_controller.rb`?

Reply:
I'm actually wondering about refactoring away the `to_account_session`s in other places, as it's only needed if the OAuth tokens change - which they don't always.  But I'm not sure if that would just add too much complexity.

Reply:
Would that be in a subsequent PR?

Reply:
I think so, I want to ruminate on it a bit.